### PR TITLE
TTD: Fix hang in pug + other NPM failures.

### PR DESCRIPTION
**Pull request by mrkmarron**

### CODE REVIEW

File: lib/Runtime/Types/DynamicObject.cpp [hunk 1]
```diff
@@ -895,6 +895,13 @@ namespace Js
 
     bool DynamicObject::TryCopy(DynamicObject* from)
     {
+#if ENABLE_TTD
+        if (from->GetScriptContext()->ShouldPerformRecordOrReplayAction())
```
Comment (MSLaguana) [inline lib/Runtime/Types/DynamicObject.cpp]: Does this essentially disable type sharing during record/replay scenarios?

Comment (mrkmarron) [reply lib/Runtime/Types/DynamicObject.cpp]: It should disable the optimization around object assignment during record and replay. Other type sharing should be enabled.

**Comment by mrkmarron** [comment]
@dotnet-bot test OSX static_osx_osx_debug
@dotnet-bot test OSX static_osx_osx_test
@dotnet-bot test Windows 10 ci_slow_x64_debug

**Comment by mrkmarron** [comment]
@dotnet-bot test Windows ci_slow_x64_debug

**Comment by kfarnung** [comment]
@mrkmarron Can we get this landed today?

**Pull request merged**